src/sql/workbench/services/notebook/common/notebookServiceImpl.ts

```diff
@@ -401,6 +401,13 @@ export class NotebookService extends Disposable implements INotebookService {
 		let providerDescriptor = this._providers.get(providerId);
 		let instance: INotebookProvider;
 
+		// Await extension registration before awaiting provider registration
+		try {
+			await this._extensionService.whenInstalledExtensionsRegistered;
```

Comment:
So... putting this here likely means SQL notebooks will be help up unnecessarily. I would suggest moving this inside the 
```
			if (!providerDescriptor.instance) {
```
block or inside the waitOnProviderAvailability method. Either one means we only await if the required object doesn't exist. For internal providers this speeds things up, for anything else it shouldn't cause any harm.